sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -188,4 +202,29 @@ class ProjectOverviewView extends VerticalLayout{
         return isAvailable
     }
 
+    private void enableWhenProjectIsSelected(Component component) {
+        viewModel.addPropertyChangeListener("selectedProject") {
+            Optional<ProjectSummary> projectSelection = Optional.ofNullable(viewModel.selectedProject)
+            component.enabled = projectSelection.present
+            component.visible = projectSelection.present
+        }
+    }
+
+    private void changeSubscriberStatus(Boolean isSubscribed) {
+        try {
+            //ToDo This will be removed once the controller/UseCase/Presenter is in place
+            viewModel.projectIsSubscribed = !isSubscribed
+            if (isSubscribed) {
+                println("Insert SubscriptionController.unsubscribe call here")
+            } else {
+                println("Insert SubscriptionController.subscribe call here")
+            }
+        }
+        catch (IllegalArgumentException illegalArgument) {
+            notificationService.publishFailure("Change in subscription status failed due to: ${illegalArgument.getMessage()}")
+        } catch (Exception ignored) {
+            notificationService.publishFailure("Subscription status change failed for unknown reasons. ${Constants.CONTACT_HELPDESK}")
+        }
+
+    }
```

Comment:
same as with the variable you added. This should be added when this user story is implemented.

Reply:
👍